Source/WebCore/layout/formattingContexts/FormattingConstraints.h

```diff
@@ -36,6 +36,7 @@ struct HorizontalConstraints {
 
     LayoutUnit logicalLeft;
     LayoutUnit logicalWidth;
+    std::optional<LayoutUnit> balanceWidth { std::nullopt };
```

Comment:
maybe this should go to InlineLayoutState to ensure HorizontalConstraints remains generic.